src/routing_table/xorable.rs

```diff
@@ -128,6 +162,22 @@ macro_rules! impl_xorable {
                 let pow_i = 1 << (mem::size_of::<Self>() * 8 - 1 - i); // 1 on bit i.
                 self ^ pow_i
             }
+
+            fn binary(&self) -> String {
+                let bits = mem::size_of::<Self>() * 8;
+                let binary = format!("{:1b}", self);
```

Comment:
If the format string were already `"{:08b}"`, `"{:016b}"`, etc., the string would already be padded with zeros. Maybe it would be easier to just `match mem::size_of::<Self>()` here instead of padding manually?